routing/dht/routing.go

```diff
@@ -40,19 +40,24 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error
 		return err
 	}
 
-	peers := dht.routingTable.NearestPeers(kb.ConvertKey(key), KValue)
-
-	query := newQuery(key, dht.network, func(ctx context.Context, p peer.ID) (*dhtQueryResult, error) {
```

Comment:
Ah, didn't see `getClosestPeers` uses a query inside. this may be ok. I still find the query + channel + waitgroup combination more confusing than it may need to be, but if this is correct, these upgrades may be worth doing separately